spec/unit/Adb.spec.js

```diff
@@ -0,0 +1,222 @@
+        it('should call adb uninstall with the correct arguments', () => {
+            const packageId = 'io.cordova.test';
+            spawnSpy.and.returnValue(Promise.resolve(''));
+
+            return Adb.uninstall(deviceId, packageId).then(() => {
+                const args = spawnSpy.calls.argsFor(0);
+                expect(args[0]).toBe('adb');
+
+                const adbArgs = args[1];
+                expect(adbArgs).toContain('uninstall');
+                expect(adbArgs.join(' ')).toMatch(`-s ${deviceId}`);
+                expect(adbArgs[adbArgs.length - 1]).toBe(packageId);
+            });
+        });
+    });
+
+    describe('shell', () => {
+        const shellCommand = 'ls -l /sdcard';
+
+        it('should run the passed command on the target device', () => {
+            spawnSpy.and.returnValue(Promise.resolve(''));
+
+            return Adb.shell(deviceId, shellCommand).then(() => {
+                const args = spawnSpy.calls.argsFor(0);
+                expect(args[0]).toBe('adb');
+
+                const adbArgs = args[1].join(' ');
+                expect(adbArgs).toContain('shell');
+                expect(adbArgs).toMatch(`-s ${deviceId}`);
+                expect(adbArgs).toMatch(new RegExp(`${shellCommand}$`));
+            });
+        });
+
+        it('should reject with a CordovaError on failure', () => {
+            const errorMessage = 'shell error';
+            spawnSpy.and.returnValue(Promise.reject(errorMessage));
+
+            return Adb.shell(deviceId, shellCommand)
+                .then(() => fail('Unexpectedly resolved'))
+                .catch(err => {
+                    expect(err instanceof Adb.__get__('CordovaError')).toBe(true);
+                    expect(err).toMatch(errorMessage);
+                });
+        });
+    });
+
+    describe('start', () => {
+        const activityName = 'io.cordova.test/.MainActivity';
+
+        it('should start an activity using the shell activity manager', () => {
+            const shellSpy = spyOn(Adb, 'shell').and.returnValue(Promise.resolve(''));
+
+            return Adb.start(deviceId, activityName).then(() => {
+                expect(shellSpy).toHaveBeenCalled();
+
+                const [target, command] = shellSpy.calls.argsFor(0);
+                expect(target).toBe(deviceId);
+                expect(command).toMatch('am start');
+                expect(command).toMatch(`-n${activityName}`);
+            });
+        });
+
+        it('should reject with a CordovaError on a shell error', () => {
+            const errorMessage = 'Test Start error';
+            spyOn(Adb, 'shell').and.returnValue(Promise.reject(errorMessage));
+
+            return Adb.start(deviceId, activityName)
+                .then(() => fail('Unexpectedly resolved'))
+                .catch(err => {
+                    expect(err instanceof Adb.__get__('CordovaError')).toBe(true);
+                    expect(err).toMatch(errorMessage);
```

Comment:
I usually explicitly check the message property to make sure it's OK. Coercing an error to string can give you all kinds of junk in addition to the message. Apply elsewhere accordingly.